### PR TITLE
build: update to 1ES PT

### PR DESCRIPTION
**Pull request by hallvictoria**
## Description

Moving to 1ES PT.
Added new pipelines referencing 1ES pipeline templates and using 1ES hosted pools.
- code mirror pipeline
- official build
- public build
- E2E tests

---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: eng/ci/official-build.yml
Comment (manvkaur) [inline eng/ci/official-build.yml]: nit: newline should be removed at L36

File: eng/ci/official-build.yml
Comment (manvkaur) [inline eng/ci/official-build.yml]: nit: newline should be removed at L42

File: eng/ci/public-build.yml
Comment (manvkaur) [inline eng/ci/public-build.yml:33]: do we not need any other steps apart from 1ES PT? Are we keeping them in github?

File: eng/templates/official/jobs/ci-e2e-tests.yml
Comment (manvkaur) [inline eng/templates/official/jobs/ci-e2e-tests.yml]: entire file - this format/syntax will probably not work with ADO, it would need conversion from Github action to ADO pipeline

**Review by manvkaur — CHANGES_REQUESTED** [review]

**Comment by manvkaur** [comment]
Please also add link to public pipeline that is running.

File: eng/ci/public-build.yml
Comment (hallvictoria) [reply eng/ci/public-build.yml:33]: Added the build stage to public-build

File: eng/templates/official/jobs/ci-e2e-tests.yml
Comment (hallvictoria) [reply eng/templates/official/jobs/ci-e2e-tests.yml]: Updated

File: eng/templates/official/jobs/build-artifacts.yml
Comment (manvkaur) [inline eng/templates/official/jobs/build-artifacts.yml:17]: is this required for public build as we share the template between both?

File: .github/workflows/ci_e2e_workflow.yml
Comment (manvkaur) [inline .github/workflows/ci_e2e_workflow.yml:64]: keep a ToDo task in your notes or ADO task to remove this file and pipeline once ADO pipelines are successfully working and added to repo rulesets

File: eng/templates/official/jobs/build-artifacts.yml
Comment (hallvictoria) [reply eng/templates/official/jobs/build-artifacts.yml:17]: Removed

File: eng/ci/public-build.yml
Comment (manvkaur) [inline eng/ci/public-build.yml]: it seems we don't need official folder as we share templates between official and public

File: .github/workflows/ci_e2e_workflow.yml
Comment (hallvictoria) [reply .github/workflows/ci_e2e_workflow.yml:64]: Added task in planning repo

File: eng/templates/official/jobs/build-artifacts.yml
Comment (manvkaur) [reply eng/templates/official/jobs/build-artifacts.yml:17]: do we also need L16 for publishVstsFeed?

File: eng/ci/public-build.yml
Comment (hallvictoria) [reply eng/ci/public-build.yml]: Good point, removed folder and refactored references

**Review by manvkaur — APPROVED** [review]
approving with suggestion so that you are not blocked on me to create the pipelines, just take a look at publishvstsfeed, if it is needed in common template

File: eng/ci/official-build.yml
Comment (gavin-aguiar) [inline eng/ci/official-build.yml]: Shouldn't this be dev

File: eng/ci/public-build.yml
Comment (gavin-aguiar) [inline eng/ci/public-build.yml]: dev here too?

File: eng/templates/official/jobs/build-artifacts.yml
Comment (hallvictoria) [reply eng/templates/official/jobs/build-artifacts.yml:17]: Updated. Publishing is only done in the official build, not for public build.

File: eng/ci/official-build.yml
Comment (hallvictoria) [reply eng/ci/official-build.yml]: Updated

File: eng/templates/jobs/build.yml
Comment (gavin-aguiar) [inline eng/templates/jobs/build.yml]: If this is running on windows it should be .env\Scripts\Activate.ps1

File: eng/ci/official-build.yml
Comment (gavin-aguiar) [inline eng/ci/official-build.yml:16]: Do we need to run official builds on a schedule or only on a trigger when something is merged to dev?

File: eng/templates/official/jobs/ci-e2e-tests.yml
Comment (gavin-aguiar) [inline eng/templates/official/jobs/ci-e2e-tests.yml]: Dotnet 8?

Comment (hallvictoria) [reply eng/templates/official/jobs/ci-e2e-tests.yml]: Updated

File: eng/ci/official-build.yml
Comment (hallvictoria) [reply eng/ci/official-build.yml:16]: Official builds need an at-least-weekly schedule for SDL reporting

**Review by gavin-aguiar — APPROVED** [review]

**Pull request merged**
